pkgs/development/compilers/llvm/12/lldb.nix

```diff
@@ -0,0 +1,99 @@
+{ lib, stdenv
+, fetch
+, cmake
+, zlib
+, ncurses
+, swig
+, which
+, libedit
+, libxml2
+, llvm
+, clang-unwrapped
+, python3
+, version
+, darwin
+, lit
+, enableManpages ? false
+}:
+
+stdenv.mkDerivation (rec {
+  pname = "lldb";
+  inherit version;
+
+  src = fetch pname "077fli9l0fg4kpa5l9vrj810s13ajs15d745lg4l8kmjkbw7p3yh";
+
+  patches = [ ./lldb-procfs.patch ];
+
+  nativeBuildInputs = [ cmake python3 which swig lit ]
+    ++ lib.optionals enableManpages [ python3.pkgs.sphinx python3.pkgs.recommonmark ];
+
+  buildInputs = [
+    ncurses
+    zlib
+    libedit
+    libxml2
+    llvm
+  ]
+  ++ lib.optionals stdenv.isDarwin [
+    darwin.libobjc
+    darwin.apple_sdk.libs.xpc
+    darwin.apple_sdk.frameworks.Foundation
+    darwin.bootstrap_cmds
+    darwin.apple_sdk.frameworks.Carbon
+    darwin.apple_sdk.frameworks.Cocoa
+  ];
```

Comment:
```suggestion
  ++ lib.optionals stdenv.isDarwin [
    libobjc
    xpc
    Foundation
    bootstrap_cmds
    Carbon
    Cocoa
  ];
```

Reply:
@SuperSandro2000 what's the reasoning for this btw? IMO this makes it worse as it isn't clear if they are part of `all-packages.nix` or inherited (e.g. form the `darwin` attrset). And this suggested change is incomplete, breaking the evaluation: https://gist.github.com/GrahamcOfBorg/285deb2c69282dad18527d1a851e614f

Reply:
darwin should be part of the inputs. The packages should be inherited one by one. Otherwise it is not that easy to overwrite individual packages.

I think I did not add the inherited part to my suggestions.

Reply:
> Otherwise it is not that easy to overwrite individual packages.

Ok, that does seem to make sense. https://github.com/NixOS/nixpkgs/pull/116646/commits/a5fdc785ea5404dd67e4c1288aa56ecf88545f66 should fix the evaluation.